Intersect.Client/Core/Graphics.cs

```diff
@@ -541,53 +553,13 @@ private static void DrawOverheadInfo(Entity entity)
                 }
             }
 
-            // Events have their own handler for hiding names within the DrawName virtual void in the Entity class.
-            if (isEvent)
-            {
-                entity.DrawName(null);
-            }
-
-            // If MyOverheadInfo is toggled on always draw the local player's info.
-            else if (Globals.Database.MyOverheadInfo && isMe)
-            {
-                entity.DrawName(null);
-            }
-
-            // If NpcOverheadInfo is toggled on, always draw npc names.
-            else if (Globals.Database.NpcOverheadInfo && isNpc)
-            {
-                entity.DrawName(null);
-            }
-
-            // If PlayerOverheadInfo is toggled on, always draw other player's info.
-            else if (Globals.Database.PlayerOverheadInfo && isOtherPlayer &&
-                !isFriend && !isGuildMate && !isPartyMate)
-            {
-                entity.DrawName(null);
-            }
-
-            // If PartyMemberOverheadInfo is toggled on, always draw party members info.
-            else if (Globals.Database.PartyMemberOverheadInfo && isPartyMate)
-            {
-                entity.DrawName(null);
-            }
-
-            // If FriendOverheadInfo & GuildMemberOverheadInfo are on, let's prevent double draw.
-            else if (Globals.Database.FriendOverheadInfo && isFriend &&
-                Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
-            {
-                entity.DrawName(null);
-                return;
-            }
-
-            // If FriendOverheadInfo is toggled on, always draw friend's info.
-            else if (Globals.Database.FriendOverheadInfo && isFriend && !isPartyMate)
-            {
-                entity.DrawName(null);
-            }
-
-            // If GuildMemberOverheadInfo is toggled on, always draw guild members info.
-            else if (Globals.Database.GuildMemberOverheadInfo && isGuildMate && !isPartyMate)
```

Comment:
This is not.

Use the old code and only compute the booleans as close to their using code as needed.

Use return statements as needed to break up the else-if chain into just ifs and comparisons as close the the consuming if statements as possible.